db-commons/src/main/scala/org/bitcoins/db/AppConfig.scala

```diff
@@ -159,31 +159,110 @@ abstract class AppConfig extends StartStopAsync[Unit] with BitcoinSLogger {
     baseDatadir.resolve(lastDirname)
   }
 
+  lazy val driver: DatabaseDriver = {
+    val driverName =
+      getConfigString(s"bitcoin-s.$moduleName.db.driverName")
+    DatabaseDriver.fromString(driverName)
+  }
+
+  lazy val jdbcUrl: String = {
+    driver match {
+      case SQLite =>
+        s""""jdbc:sqlite:"$dbPath/$dbName"""
+      case PostgreSQL =>
+        s""""jdbc:postgresql://$dbHost:$dbPort/postgres""""
+    }
+  }
+
+  /** The path where our DB is located */
+  lazy val dbPath: Path = {
+    val pathStrOpt =
+      getConfigStringOpt(s"bitcoin-s.$moduleName.db.path")
+    pathStrOpt match {
+      case Some(pathStr) =>
+        Paths.get(pathStr)
+      case None =>
+        sys.error(s"Could not find dbPath for $moduleName.db.path")
+    }
+  }
+
+  /** The name of our database */
+  lazy val dbName: String = {
+    val name = getConfigString(s"bitcoin-s.$moduleName.db.name")
+
+    driver match {
+      case SQLite     => s"$name.sqlite"
+      case PostgreSQL => name
+    }
+  }
+
+  /** The host of our postgresql database */
+  lazy val dbHost: String = {
+    val hostOpt = getConfigStringOpt(s"bitcoin-s.$moduleName.db.host")
+
+    hostOpt match {
+      case Some(host) => host
+      case None       => "localhost"
+    }
+  }
+
+  /** The port number of our postgresql database */
+  lazy val dbPort: Int = {
+    val portOpt = config.getIntOpt(s"bitcoin-s.$moduleName.db.port")
+
+    portOpt match {
+      case Some(port) => port
+      case None       => 5432 // default postgres port
+    }
+  }
+
+  lazy val schemaName: Option[String] = driver match {
+    case PostgreSQL =>
+      Some(moduleName)
+    case SQLite =>
+      None
+  }
+
   lazy val slickDbConfig: DatabaseConfig[JdbcProfile] = {
+    // Create overrides if modules want to change their path or db name
+    val overrideConf = ConfigFactory.parseString {
+      s"""
+         |bitcoin-s {
+         |  $moduleName {
```

Comment:
Hmmm. I'm not sure if I understand what this is for. 

Shouldn't this allow just be put into the configuration files?

Reply:
Interesting, so you are saying that our configuration can be internally inconsistent when we specify `walletName`?

https://bitcoin-s.org/docs/next/config/configuration#database-configuration

Reply:
Yeah, because we need to add on the wallet name to the dbPath